dexmaker/src/test/java/com/google/dexmaker/stock/ProxyBuilderTest.java

```diff
@@ -879,4 +881,70 @@ public void setFakeResult(Object result) {
             fakeResult = result;
         }
     }
+
+    public static class TestOrderingClass {
+        public int returnsInt() {
+            return 0;
+        }
+
+        public int returnsInt(int param1, int param2) {
+            return 1;
+        }
+
+        public String returnsString() {
+            return "string";
+        }
+
+        public boolean returnsBoolean() {
+            return false;
+        }
+
+        public double returnsDouble() {
+            return 1.0;
+        }
+
+        public Object returnsObject() {
+            return new Object();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testMethodsGeneratedInDeterministicOrder() throws Exception {
+        Field mapField = ProxyBuilder.class
+                .getDeclaredField("generatedProxyClasses");
+        mapField.setAccessible(true);
+
+        // Grab the static methods array from the original class.
+        Method[] methods1 = getMethodsForProxyClass(TestOrderingClass.class);
+        assertNotNull(methods1);
+
+        // Clear ProxyBuilder's in-memory cache of classes. This will force
+        // it to rebuild the class and reset the static methods field.
+        Map<Class<?>, Class<?>> map = (Map<Class<?>, Class<?>>) mapField
+                .get(null);
+        assertNotNull(map);
+        map.clear();
```

Comment:
Update:  discussed in person - we agreed not to add special, test-only methods to ProxyBuilder.